WooCommerce/WooCommerceScreenshots/Screens/MyStoreScreen.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import XCTest
+
+class MyStoreScreen: BaseScreen {
+
+    struct ElementStringIDs {
+        static let topBannerCloseButton = "top-banner-view-dismiss-button"
+        static let settingsButton = "dashboard-settings-button"
+    }
+
+    let tabBar = TabNavComponent()
+    let periodStatsTable = PeriodStatsTable()
+    let settingsButton: XCUIElement
+    let topBannerCloseButton: XCUIElement
+
+    static var isVisible: Bool {
+        let settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
+        return settingsButton.exists && settingsButton.isHittable
+    }
+
+    init() {
+        settingsButton = XCUIApplication().buttons[ElementStringIDs.topBannerCloseButton]
+        topBannerCloseButton = XCUIApplication().buttons[ElementStringIDs.topBannerCloseButton]
+
+        super.init(element: settingsButton)
+    }
+
+    func dismissTopBannerIfNeeded() -> MyStoreScreen {
+        topBannerCloseButton.tap()
```

Comment:
We should check for existence of the top banner before trying to dismiss it. (I used a test site that doesn't have WC Admin installed, so there was no banner to dismiss.)

This isn't a huge deal if we plan to always use the same store for test runs, and since we'll be mocking network requests eventually, but the "IfNeeded" part of `dismissTopBannerIfNeeded` makes me want this. :)

Reply:
Definitely!! This is changed!